partition_table.go

```diff
@@ -480,30 +442,31 @@ func (p *PartitionTable) drainConsumer(cons sarama.PartitionConsumer, errs *mult
 		}
 	})
 
-	errg.Wait()
+	return errg.Wait().ErrorOrNil()
 }
 
-func (p *PartitionTable) loadMessages(ctx context.Context, cons sarama.PartitionConsumer, partitionHwm int64, stopAfterCatchup bool) (rerr error) {
-	errs := new(multierr.Errors)
-
-	// deferred error handling
-	defer func() {
-		errs.Collect(rerr)
-
-		rerr = errs.NilOrError()
-		return
-	}()
+func (p *PartitionTable) loadMessages(ctx context.Context, cons sarama.PartitionConsumer, partitionHwm int64, stopAfterCatchup bool) error {
 
 	stallTicker := time.NewTicker(p.stallPeriod)
 	defer stallTicker.Stop()
 
 	lastMessage := time.Now()
 
+	messages := cons.Messages()
+	errors := cons.Errors()
+
 	for {
 		select {
-		case msg, ok := <-cons.Messages():
+		case err, ok := <-errors:
```

Comment:
I would use `more` instead of `ok` when checking if channels are open.
Then you can write
```
if !more {
    // stop processing
}
```